src/components/Network.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect } from "react";
+import { useLocation } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import styled from "styled-components";
+import { update } from "ducks/settings";
+import { useRedux } from "hooks/useRedux";
+
+const BannerEl = styled.div`
+  width: 100%;
+  padding: 10px 0;
+  text-align: center;
+  background-color: #c00;
+  color: #fff;
+`;
+
+interface NetworkProps {
+  children: React.ReactNode;
+}
+
+export const Network = ({ children }: NetworkProps) => {
+  const dispatch = useDispatch();
+  const { settings } = useRedux(["settings"]);
+  const queryParams = new URLSearchParams(useLocation().search);
+  const testnetParam = queryParams.get("testnet");
+
+  const isDevelopment = process.env.NODE_ENV === "development";
+  const isTestnet = testnetParam === "true";
+  const showNetworkMessage =
+    (isDevelopment && !settings.isTestnet) ||
+    (!isDevelopment && settings.isTestnet);
+
+  const handleTestnetParamChange = () => {
+    if (testnetParam) {
+      // route and store will reset when page reloads on query change
+      dispatch(update({ isTestnet }));
+    }
+  };
+
+  useEffect(handleTestnetParamChange, [isTestnet]);
```

Comment:
I'm finding it a little confusing to represent the useEffect callback this way. I expect event handlers to be named `handle{Action}`. Seems better just to implement it inline.

Reply:
This is one is a weird React bug where it complains that the called function is not passed as a dependency. In this case it is `dispatch()`: https://github.com/facebook/react/issues/15865. We could disable this rule globally, but there might be legit cases where it might help. Instead of adding `eslint-disable` to every place where this is an issue, we decided to create an external function. Should we do that instead?

Reply:
For that, I would just pass `dispatch` as a dependency rather than try to ignore it. Ostensibly it won't have any effect on the component, since the `dispatch` function probably won't change.

Reply:
I'll create a separate PR to update this throughout the app.